admin/fleetcommander/utils.py

```diff
@@ -115,3 +116,15 @@ def merge_settings(a, b):
         result[domain] = [index[key] for key in index]
 
     return result
+
+
+def get_free_port():
```

Comment:
Following up from our conversation on IRC, Soup.Server takes 0 as an argument as well and this function is quite undeterministic, consumes time and is the equivalent of using 0 as the port in the Soup.Server class.

I don't understand why we have to maintain such code if the main codepath already takes 0 as a port.

You can access the "port" property of the Soup server once it's connected to get the port back.